project/celery.py

```diff
@@ -16,17 +18,47 @@
 
 app.config_from_object("django.conf:settings", namespace="CELERY")
 
-app.conf.update(
-    # Hacks to work around the lack of built in support for Redis cluster.
-    # We don't use a cluster directly but Azure Managed Redis acts like one.
-    # https://github.com/celery/celery/issues/8276#issuecomment-2082176893
-    broker_transport_options={
-        "global_keyprefix": "{queue}:"
-    },
-    result_backend_transport_options={
-        "global_keyprefix": "{rest}:"
-    }
-)
+# Default Redis configuration for local development
+broker_url = os.environ.get('CELERY_BROKER_URL')
+result_backend = os.environ.get('CELERY_RESULT_BACKEND')
+
+# Local development Redis configuration
+REDIS_HOSTNAME = os.environ.get('REDIS_HOSTNAME')  
+REDIS_PORT = os.environ.get('REDIS_PORT', '6379')
+REDIS_PASSWORD = os.environ.get('REDIS_PASSWORD')  
+
+broker_url = f"redis://{':' + REDIS_PASSWORD + '@' if REDIS_PASSWORD else ''}{REDIS_HOSTNAME}:{REDIS_PORT}/0"
+result_backend = f"redis://{':' + REDIS_PASSWORD + '@' if REDIS_PASSWORD else ''}{REDIS_HOSTNAME}:{REDIS_PORT}/1"
+
+
+# Conditional configuration for Azure environment
+if "AZURE_CONTAINER_ENVIRONMENT" in os.environ:
```

Comment:
If we make this our own setting in `.env` we can conditionally enable in staging